src/FileServiceManager/FileService/DAL/FileSystemConfiguration/FileSystemJsonConfiguration/FileSystemJsonConfigurator.cs

```diff
@@ -0,0 +1,206 @@
+                        compArray.Add(JToken.FromObject(company));
+                    }
+
+                    using (fileWriter = File.CreateText(_configPath))
+                    {
+                        using (JsonTextWriter writer = new JsonTextWriter(fileWriter) { Formatting = Formatting.Indented })
+                        {
+                            await rootToken.WriteToAsync(writer);
+                        }
+                    }
+                }
+                else
+                {
+                    throw new FileNotFoundException("Configuration file is not found");
+                }
+            }
+
+            catch (Exception)
+            {
+                throw;
+            }
+            finally
+            {
+                fileWriter?.Close();
+            }
+        }
+
+        public async IAsyncEnumerable<FileMetadata> GetAllConfigurationsAsync(int companyId)
+        {
+            JToken filesToken;
+            try
+            {
+                JToken rootToken = await GetRootTokenAsync();
+
+                if (rootToken != null)
+                {
+                    filesToken = rootToken.SelectToken($"$.companies[?(@.companyId == {companyId})]")?.SelectToken($"$..files");
+                }
+                else
+                {
+                     throw new FileNotFoundException("Configuration file is not found");
+                }
+            }
+
+            catch (Exception)
+            {
+                throw;
+            }
+
+            if (filesToken == null)
+            {
+                throw new InvalidOperationException($"Company ID = {companyId} is not found");
+            }
+            else
+            {
+                foreach (var fileToken in filesToken)
+                {
+                    yield return fileToken.ToObject<FileMetadata>();
+                }
+            }
+        }
+
+        public async Task<FileMetadata> GetConfigurationAsync(int companyId, Guid fileId)
+        {
+            try
+            {
+                JToken rootToken = await GetRootTokenAsync();
+
+                if (rootToken != null)
+                {
+                    JToken filesToken = rootToken.SelectToken($"$.companies[?(@.companyId == {companyId})]")?.SelectToken($"$..files[?(@.Id == '{fileId}')]");
+
+                    if (filesToken != null)
+                    {
+                        return filesToken.ToObject<FileMetadata>();
+                    }
+                    else
+                    {
+                        throw new InvalidOperationException($"File for Company ID = { companyId } is not found");
+                    }
+                }
+                else
+                {
+                    throw new FileNotFoundException("Configuration file is not found");
+                }
+            }
+
+            catch (Exception)
+            {
```

Comment:
what is the meaning of this ?

Reply:
It was remained from previous changes, I make code refactoring.